src/framework/constraints.ml

```diff
@@ -755,8 +755,31 @@ struct
   let system (v,c) =
     match v with
     | FunctionEntry _ when full_context ->
-      [fun _ _ _ _ -> S.val_of c]
-    | _ -> List.map (tf (v,c)) (Cfg.prev v)
+      Some (fun _ _ _ _ -> S.val_of c)
+    | FunctionEntry _ ->
+      None
+    | _ ->
+      let tf getl sidel getg sideg =
+        let tf' eu = tf (v,c) eu getl sidel getg sideg in
+
+        match NodeH.find_option CfgTools.node_scc_global v with
+        | Some scc when NodeH.mem scc.prev v ->
+          let stricts = NodeH.find_all scc.prev v in
+          let xs_stricts = List.map tf' stricts in
+          if List.for_all S.D.is_bot xs_stricts then
+            S.D.bot ()
+          else
+            let xs_strict = List.fold_left S.D.join (S.D.bot ()) xs_stricts in
```

Comment:
Could we not just compute `xs_strict` and check that for bottom? Might be more efficient since we expect it to be non-bottom most of the time (and in this case we only go over the list once)

Also, it simplifies the code a bit

Reply:
I guess joining a bunch of DeadCode bottoms should also be fairly cheap, but you're right, probably not really worth benchmarking.